app/models/user.rb

```diff
@@ -11,10 +9,43 @@ class User < ActiveRecord::Base
          :trackable,
          :validatable
 
+  after_create :send_admin_mail
+
   belongs_to :agency
 
-  scope :case_workers_by_agency, ->(agency = nil) { case_workers.where(agency: agency) }
-  scope :case_workers, -> { where(role: :case_worker) }
+  def self.case_workers_by_agency(agency = nil)
+    case_workers.where(agency: agency)
+  end
+
+  def self.case_workers
+    where(role: :case_worker)
+  end
+
+  def self.send_reset_password_instructions(attributes = {})
```

Comment:
You could override a more low-level method here:


        def send_reset_password_instructions_notification(token)
          send_devise_notification(:reset_password_instructions, token, {})
        end

This could be overridden as:

        def send_reset_password_instructions_notification(token)
            approved? ? super : errors[:base] << I18n.t("devise.failure.not_approved")
        end